### PR TITLE
Update to Cranelift 0.3.0

### PR DESCRIPTION
**Pull request by bjfish**
Some API changes were addressed:
- `ModuleEnvironment` many trait methods removed: https://github.com/CraneStation/cranelift/commit/459f6ddb268bdcfc7bf23692b7ff1abf5a2fea3e#diff-0682a088315ee358a75d4b60bc72dafc
- `declare_table_elements` parameter type changed: https://github.com/CraneStation/cranelift/commit/64ea964bc0b6f985de4064393f5652cee8fdb812
- `UnreachableCodeReached` trap code added: https://github.com/CraneStation/cranelift/commit/1c43ad771e33fab3b52088b5792f7c07369a986a
- `declare_signature` signature updated: https://github.com/CraneStation/cranelift/commit/8a24539e0c5fa5af62d4894760ee2a7ca8656ba9
- `define_function_body` parameter added: https://github.com/CraneStation/cranelift/commit/37dffdaea1a3fefc0374be2d2c154cdc96217711
- `target_lexicon` dependency updated to version `0.3.0`

### CODE REVIEW

File: lib/clif-backend/src/module_env.rs [hunk 1]
```diff
@@ -50,6 +50,20 @@ impl<'module, 'isa> ModuleEnv<'module, 'isa> {
...
+    }
+
+    /// Return the signature index for the given function index.
+    pub fn get_func_type(
+        &self,
+        func_index: cranelift_wasm::FuncIndex,
+    ) -> cranelift_wasm::SignatureIndex {
+        let sig_index: SigIndex = self.module.info.func_assoc[Converter(func_index).into()];
+        Converter(sig_index).into()
+    }
```
Comment (bjfish) [inline lib/clif-backend/src/module_env.rs]: These were the only two of the removed trait methods that were used within our codebase, so I added them here.

**Comment by bjfish** [comment]
bors try

File: lib/clif-backend/src/module.rs [hunk 2]
```diff
@@ -151,8 +151,8 @@ convert_clif_to_runtime_index![
     (SignatureIndex: SigIndex),
 ];
 
-impl<'a> From<Converter<&'a ir::Signature>> for FuncSig {
-    fn from(signature: Converter<&'a ir::Signature>) -> Self {
+impl<'a> From<Converter<ir::Signature>> for FuncSig {
```
Comment (syrusakbary) [inline lib/clif-backend/src/module.rs]: Perhaps we can remove `<'a>` ?

```suggestion
impl From<Converter<ir::Signature>> for FuncSig {
```

**Comment by bjfish** [comment]
bors try

File: lib/clif-backend/src/module.rs (hunk 2 above)
Comment (bjfish) [reply lib/clif-backend/src/module.rs]: Good catch, I committed the suggestion.

**Comment by bjfish** [comment]
bors try

**Pull request merged**